### PR TITLE
calculate_production.py

### PR DESCRIPTION
**Pull request by Grummel7**
More preparation. Surprising many things were missing in the old code. Do I still miss some bonuses?

### CODE REVIEW

File: default/python/AI/AIDependencies.py [hunk 1]
```diff
@@ -216,6 +216,21 @@ class Tags:
         metabolismBoosts[boost] = metab
 # </editor-fold>
 
+# <editor-fold desc="Industry boosting specials">
+# modified = affected by species multiplier
+industry_boost_specials_modified = {
```
Comment (Cjkjvfnby) [inline default/python/AI/AIDependencies.py:221]: Maybe use `fixed`/`flat` and `multiplier` terms?

File: default/python/AI/colonization/calculate_production.py [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+    # TBD check connection!?
+    # TBD: cache bonuses?
+    have_center = bool(BuildingType.INDUSTRY_CENTER.built_or_queued_at())
+    centre_bonus1 = get_named_real("BLD_INDUSTRY_CENTER_1_TARGET_INDUSTRY_PERPOP")
+    centre_bonus2 = get_named_real("BLD_INDUSTRY_CENTER_2_TARGET_INDUSTRY_PERPOP")
+    centre_bonus3 = get_named_real("BLD_INDUSTRY_CENTER_3_TARGET_INDUSTRY_PERPOP")
+    bonuses = [
+        _ProductionBonus(
+            tech_is_complete("PRO_FUSION_GEN") or "PRO_FUSION_GEN" in research_list[:3],
```
Comment (Cjkjvfnby) [inline default/python/AI/colonization/calculate_production.py]: It sounds like a candidate for extraction to a separate method.

File: default/python/AI/colonization/calculate_production.py [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+class _ProductionBonus(NamedTuple):
+    available: bool
+    min_stability: float
+    value: float
+
+
+def _get_production_bonus_modified(planet: fo.planet, stability: float, research_list: Iterable):
+    """
+    Calculate bonus production per population which is added before multiplication with the species skill value.
```
Comment (Cjkjvfnby) [inline default/python/AI/colonization/calculate_production.py]: I'd say it's not calculating a real bonus, but it predicts future bonuses.  Let's reflect this in the docs.

**Review by Cjkjvfnby — COMMENTED** [review]
Left a couple of cosmetic comments.

File: default/python/AI/colonization/calculate_production.py (hunk 2 above)
Comment (Grummel7) [reply default/python/AI/colonization/calculate_production.py]: Well, we not, then I can also cache it.

**Review by Cjkjvfnby — APPROVED** [review]

**Pull request merged**